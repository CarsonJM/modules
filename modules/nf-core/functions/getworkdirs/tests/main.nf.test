nextflow_function {

    name "Test getWorkDirs"
    script "../main.nf"

    test("getWorkDirs single") {
        function "getWorkDirs"
        when {
            function {
            """
            input[0] = Channel.of(
                [
                    [ id:'test' ], // meta map
                    "/somepath/work/12/hashforworkdirtoclean/somefile.txt"
                ]
            )
            input[1] = Channel.of(
                [
                    [ id:'test' ], // meta map
                    "/somepath/work/34/hashforworkdirdownstream/someotherfile.txt"
                ]
            )
            input[2] = true
            """
            }
        }
        then {
            assertAll (
                { assert function.success },
                { assert snapshot(function.stdout).match() }
            )
        }
    }

    test("getWorkDirs double") {
        function "getWorkDirs"
        when {
            function {
            """
            input[0] = Channel.of(
                [
                    [ id:'test' ], // meta map
                    "/somepath/work/12/hashforworkdirtoclean/somefile.txt"
                ],
                [
                    [ id:'test' ], // meta map
                    "/somepath/work/24/hashforworkdirtoclean/somefile.txt"
                ]
            )
            input[1] = Channel.of(
                [
                    [ id:'test' ], // meta map
                    "/somepath/work/34/hashforworkdirdownstream/someotherfile.txt"
                ]
            )
            input[2] = true
            """
            }
        }
        then {
            assertAll (
                { assert function.success },
                { assert snapshot(function.stdout).match() }
            )
        }
    }
}