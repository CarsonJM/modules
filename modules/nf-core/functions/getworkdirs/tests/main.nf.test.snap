{
    "getWorkDirs double": {
        "content": [
            [
                "[[id:test], /somepath/work/12/hashforworkdirtoclean/]",
                "[[id:test], /somepath/work/24/hashforworkdirtoclean/]"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-02T08:00:08.332845613"
    },
    "getWorkDirs single": {
        "content": [
            [
                "[[id:test], /somepath/work/12/hashforworkdirtoclean/]"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-02T08:00:03.219909745"
    }
}